src/hotspot/share/gc/g1/g1ParallelCleaning.cpp

```diff
@@ -32,15 +32,15 @@
 
 #if INCLUDE_JVMCI
 JVMCICleaningTask::JVMCICleaningTask() :
-  _cleaning_claimed(0) {
+  _cleaning_claimed(false) {
 }
 
 bool JVMCICleaningTask::claim_cleaning_task() {
-  if (_cleaning_claimed) {
+  if (Atomic::load(&_cleaning_claimed)) {
     return false;
   }
 
-  return Atomic::cmpxchg(&_cleaning_claimed, 0, 1) == 0;
+  return !Atomic::cmpxchg(&_cleaning_claimed, false, true);
```

Comment:
It might be an int because making it a bool may involve a byte-sized cmpxchg, which has a not very
nice implementation on some platforms.  Though even with that the claim cost is likely dwarfed by
other costs here.

Reply:
Yeah, that's probably correct.  We used to have a cmpxchg overload for jbyte, but never explicitly bool.